app/src/main/java/com/infomaniak/mail/ui/main/menu/MailboxesAdapter.kt

```diff
@@ -92,12 +108,18 @@ class MailboxesAdapter(
     ) {
         text = mailbox.email
 
-        if (!isCurrentMailbox) {
-            setOnClickListener {
-                trackerCallback()
-                onValidMailboxClicked?.invoke(mailbox.mailboxId)
+        fun getClickListener(isCurrentMailbox: Boolean): OnClickListener? {
+            return if (!isCurrentMailbox) {
+                OnClickListener {
+                    trackerCallback()
+                    onValidMailboxClicked?.invoke(mailbox.mailboxId)
+                }
+            } else {
+                null
```

Comment:
If you're gonna make it an inner function, you don't need to pass in the `isCurrentMailbox` argument, it's already accessible inside `getClickListener()`

Reply:
I actually especially like it there cause you save yourself from passing 3 different arguments to the function